073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,150 @@
+2. The cluster and user operator, during their respective reconcile loops:
+   1. Add the CA certificate to the relevant component truststore Kubernetes Secret and change the state to TRUSTED_UNUSED
+   2. Generate end entity certificates using the CA and add the EE certificate to the relevant component keystore Kubernetes Secret and change the state to TRUSTED_IN_USE
+
+Update process (e.g. CA private key has been updated):
+1. The CaReconciler generates a new CA certificate and places it in the relevant Kubernetes Secret with the state UNTRUSTED. It also updates the old CA certificate to have the state PHASE_OUT.
+2. The cluster and user operator, during their respective reconcile loop:
+   1. Add the CA certificate to the relevant component truststore Kubernetes Secret and change the state to TRUSTED_UNUSED
+   2. Generate end entity certificates using the CA and add the EE certificate to the relevant component keystore Kubernetes Secret and change the state to TRUSTED_IN_USE
+   3. Update the old CA certificate to ZOMBIE, indicating that no EE certicates are using the CA certificate
+3. In a future loop the CaReconciler removes the CA certificate from the shared Kubernetes Secret (because it is observed to be in the `ZOMBIE` state).
+4. The cluster and user operator remove the CA certificate from the truststore Kubernetes Secret
+
+Notes:
+1. The private key for the self-signed cluster CA certificate in a Kafka cluster called **foo**, can be stored in `foo-cluster-ca`.
+2. The strimzi.io/cluster-ca-key-generation can be used by the cluster operator when managing the self-signed cluster CA certificate
+3. A new annotation strimzi.io/cluster-ca-cert-fingerprint on components can be used to store the fingerprint of the issuing CA. This removes the need for the current annotation strimzi.io/cluster-ca-cert-generation which just holds an integer.
+
+### Future changes
+
+Once this proposal is implemented, it allows in future to update to change both the issuing of end entity certificates and the management of the CA certificates to be done by certificate management solutions, similarly to the [closed proposal][pr46].
+This proposal is the first step because it decouples the management of the CA certificates, from the rolling out of trust of those CA certificates.
+The next logical step will be to change the end-entity certificate issuance process, so that an end-entity certificate and it's root CA certificate can be received at the same time.
+Currently it is assumed that the CA certificate is already issued and is used for signing the end-entity certificates, rather than an end-entity certificate has been received that has been signed by (from the PoV of the requesting operator) and arbitrary CA.
+
+## Affected/not affected projects
+
+This affects the strimzi-kafka-operator project only.
+
+## Compatibility
+
+Call out any future or backwards compatibility considerations this proposal has accounted for.
```

Comment:
I think this should explain what happens during the upgrade when using the Strimzi and when using the Custom CAs.

Reply:
@scholzj do you mean what would happen during a regular upgrade, or specifically when upgrading from Strimzi using the old mechanism, to this new mechanism?